geode-core/src/main/java/org/apache/geode/internal/net/BufferPool.java

```diff
@@ -114,6 +138,70 @@ private ByteBuffer acquireBuffer(int size, boolean send) {
     return result;
   }
 
+  private ByteBuffer acquireSmallBuffer(boolean send) {
+    // set
+    int size = Connection.SMALL_BUFFER_SIZE;
+    ByteBuffer result;
+
+    BBSoftReference ref = bufferSmallQueue.poll();
+    while (ref != null) {
+      ByteBuffer bb = ref.getBB();
+      if (bb == null) {
+
+        // it was garbage collected
+        if (ref.getSend()) { // fix bug 46773
+          stats.incSenderBufferSize(-size, true);
+        } else {
+          stats.incReceiverBufferSize(-size, true);
+        }
+      } else {
+        bb.rewind();
+        bb.limit(size);
+        return bb;
+      }
+      ref = bufferSmallQueue.poll();
+    }
+    result = ByteBuffer.allocateDirect(size);
```

Comment:
If you need to allocate then do it of the fixed size (SMALL_BUFFER_SIZE not the requested size) and then set the limit to the requested size. The stats should be updated with the allocation size, not the requested size.